### PR TITLE
Us27

### PR DESCRIPTION
**Pull request by OlenaMits**
create table for friends, create endpoints: getUserFriends, send-request, update-status, delete friend

### CODE REVIEW

File: social-network-api/src/main/java/com/facebook/controller/FriendsController.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        Long userId = currentUserService.getCurrentUserId();
+        friendsService.deleteFriend(
+                userId,
+                request.getFriendId()
+        );
+        return ResponseEntity.ok().build();
+    }
+
+    @GetMapping("/list")
+    public ResponseEntity<List<FriendsResponse>> getUserFriends() {
```
Comment (miegami) [inline social-network-api/src/main/java/com/facebook/controller/FriendsController.java:56]: Also we need the endpoint to get the friends for id.

File: social-network-api/src/main/java/com/facebook/dto/friends/FriendsResponse.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    private LocalDateTime created_date;
+
+    private LocalDateTime last_modified_date;
+
+    private Long userId;
+
+    private Long friendId;
+
+    private FriendsStatus status;
```
Comment (miegami) [inline social-network-api/src/main/java/com/facebook/dto/friends/FriendsResponse.java:21]: Why the fields are in different style? Please, use camelCase for fields, if you need to get snake_case in the json you can add the annotation @JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)

Comment (OlenaMits) [reply social-network-api/src/main/java/com/facebook/dto/friends/FriendsResponse.java:21]: Fixed)

File: social-network-api/src/main/java/com/facebook/controller/FriendsController.java (hunk 1 above)
Comment (OlenaMits) [reply social-network-api/src/main/java/com/facebook/controller/FriendsController.java:56]: This endpoint get friends by user id, who is authorised now.
Rename it.

**Review by miegami — DISMISSED** [review]

**Review by yourhostel — APPROVED** [review]

**Pull request merged**